core/templates/dev/head/components/share/SharingLinksDirective.js

```diff
@@ -22,39 +22,51 @@ oppia.directive('sharingLinks', [function() {
     scope: {
       layoutType: '@',
       layoutAlignType: '@',
+      shareType: '@',
       getTwitterText: '&twitterText',
-      getExplorationId: '&explorationId'
+      getExplorationId: '&explorationId',
+      getCollectionId: '&collectionId'
     },
     templateUrl: 'components/sharingLinks',
     controller: [
-      '$scope', '$window', 'oppiaHtmlEscaper', 'ExplorationEmbedButtonService',
+      '$scope', '$window', '$injector', 'oppiaHtmlEscaper',
       'siteAnalyticsService', 'UrlInterpolationService',
       function(
-          $scope, $window, oppiaHtmlEscaper, ExplorationEmbedButtonService,
+          $scope, $window, $injector, oppiaHtmlEscaper,
           siteAnalyticsService, UrlInterpolationService) {
-        $scope.explorationId = $scope.getExplorationId();
+        if ($scope.shareType === 'exploration') {
+          $scope.explorationId = $scope.getExplorationId();
+          var ExplorationEmbedButtonService = $injector.get(
+            'ExplorationEmbedButtonService');
 
-        $scope.registerShareExplorationEvent = function(network) {
-          siteAnalyticsService.registerShareExplorationEvent(network);
-        };
+          $scope.registerShareExplorationEvent = function(network) {
```

Comment:
Why not define a single $scope.registerShareEvent method that is null to start with, then in the if/else branching do:
`
   $scope.registerShareEvent = siteAnalyticsService.registerShareExplorationEvent
`
or
`
   $scope.registerShareEvent = siteAnalyticsService.registerShareCollectionEvent
`
as appropriate? Then you avoid branching elsewhere.